src/kernel/sporadic.c

```diff
@@ -150,7 +150,11 @@ void refill_new(sched_context_t *sc, word_t max_refills, ticks_t budget, ticks_t
     /* full budget available */
     REFILL_HEAD(sc).rAmount = budget;
     /* budget can be used from now */
-    REFILL_HEAD(sc).rTime = NODE_STATE_ON_CORE(ksCurTime, core);
+    if (period == 0) {
```

Comment:
does this change makes proof easier? I was trying to avoid adding more branches to the code - e.g if I could avoid logic like

```
if (rr) { do this } else { //sporadic do that}
```

for performance

Reply:
I think the proofs end up having to distinguish round-robin and spordaic almost entirely. This particular change may have just been to make it trivial to show that a round-robin thread was always ready by the logic of sporadic tasks.

Reply:
"may have" ??  I'm not convinced about merging this commit